arangod/RocksDBEngine/RocksDBBuilderIndex.h

```diff
@@ -0,0 +1,141 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2014-2016 ArangoDB GmbH, Cologne, Germany
+/// Copyright 2004-2014 triAGENS GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Simon Grätzer
+////////////////////////////////////////////////////////////////////////////////
+
+#ifndef ARANGOD_ROCKSDB_ENGINE_ROCKSDB_BUILDER_INDEX_H
+#define ARANGOD_ROCKSDB_ENGINE_ROCKSDB_BUILDER_INDEX_H 1
+
+#include "RocksDBEngine/RocksDBIndex.h"
+
+#include <mutex>
+
+namespace arangodb {
+
+class RocksDBBuilderIndex final : public arangodb::RocksDBIndex {
+
+ public:
+  /// @brief return a VelocyPack representation of the index
+  void toVelocyPack(velocypack::Builder& builder,
+                    std::underlying_type<Index::Serialize>::type) const override;
+  
+  char const* typeName() const override {
+    return _wrapped->typeName();
+  }
+  
+  IndexType type() const override {
+    return _wrapped->type();
+  }
+  
+  bool canBeDropped() const override {
+    return false; // TODO ?!
```

Comment:
I think this will for sure be important to add a "cancel" option when we make it asynchronous. But it is fine as is for this PR.